### ISSUE TITLE
request

### ISSUE THREAD

**Issue by porres** [pure-data/pure-data#1341]

**Comment by umlaeute** [comment]
> *we* don't want to search...

this is definitely **not true** for me.
most of the times *I* do a search in a complex big patch with zillions of abstractions for some keyword that I know is hidden somewhere deep.

that's not to say, that such a checkbox might not be useful (just that the generalisation in your pretext is wrong)

**Comment by Ant1r** [comment]
This _is_ true is these cases:
- the abstraction is GOP ( #1052 )
- the abstraction is a clone (#1044)
- the error originates from a message box (#1045)

I thins these should be considered as bugs.
Fixes for these bugs (and some others) are already proposed in #965 and #972. I've been working on a daily basis with a Pd patched with these fixes for more than a year, so I can say they are fully tested now... I'd love they were accepted one of these days, so I can use the standard Pd version available for my distribution... and anybody can benefit from these corrections.

**Comment by umlaeute** [comment]
@Ant1r i'm not sure i understand what you mean here (that is: *what* do you think is true?)

neither GOP-abstractions nor cloned abstraction are *find*able right now. i agree with you that this is a bug.

but @porres request (as I understand it) was actually to make normal (non-GOP & non-cloned) abstractions **not** findable as well.

it seems to me that you agree with me that (at least sometimes) "we *do* want to also search in abstractions".

**Comment by Ant1r** [comment]
Yep, very sorry I misread the subject... Next time I'll make sure I'm fully awake before posting ;-)
Anyway I'm glad you agree they are bugs.

So back to the subject: I agree it could be useful to have the possibility to not search in abstractions, e.g a "search only in the current patch" checkbox.

**Comment by porres** [comment]
> So back to the subject: I agree it could be useful to have the possibility to not search in abstractions, e.g a "search only in the current patch" checkbox.

could you also work on this in your current PR #965 ?

**Comment by Ant1r** [comment]
Mmm.. not sure. I'd prefer keeping #965 a bug fix only, and adding new functionalities in a new PR... after this one has been merged (or rejected...).
